src/main/kotlin/org/rust/ide/formatter/processors/RsTrailingCommaFormatProcessor.kt

```diff
@@ -84,7 +86,12 @@ fun CommaList.addTrailingCommaForElement(list: PsiElement): Boolean {
     )
     if (!trailingSpace.contains('\n')) return false
 
-    if (list.firstChild.getNextNonCommentSibling() == lastElement) return false
+    if (list.firstChild.getNextNonCommentSibling() == lastElement) {
+        // allow trailing comma only for struct block fields with a single item
```

Comment:
What means "with a single item"? I don't get it :(

Reply:
Sorry for the confusion. `RsTrailingCommaFormatProcessor` does not add trailing commas to comma lists with a single element. So the plugin would format the following struct like this:
```rust
struct S {
    a: u32
}
```
while `rustfmt` formats it like this:
```rust
struct S {
    a: u32,
}
```
so I allowed the trailing comma for a comma list with a single element, but only in structs with named fields.

To make matters more complicated, the actual behaviour of `rustfmt` is this: it adds a trailing comma for fields only if the fields are multi-line. Fields are multi-line if:
1) They are inside struct with named fields ("classic" structs)
2) They are inside tuple struct or inside an enum variant and they are too long (but this happens rarely, the fields basically need to be longer than the maximum line length).

So this condition covers case 1), which is most common.